### PR TITLE
Mark tests known to fail on 32bit architectures with xfail, backport to the v0.14.x branch

### PR DESCRIPTION
**Pull request by hmaarrfk**
@jni enjoy, I tested it with our emergency 32bit instructions. Should have a high chance of passing.

### CODE REVIEW

File: skimage/_shared/testing.py [hunk 1]
```diff
@@ -26,10 +27,14 @@
 SKIP_RE = re.compile("(\s*>>>.*?)(\s*)#\s*skip\s+if\s+(.*)$")
 
 skipif = pytest.mark.skipif
+xfail = pytest.mark.xfail
 parametrize = pytest.mark.parametrize
 raises = pytest.raises
 fixture = pytest.fixture
 
+# true if python is running in 32bit mode
```
Comment (jni) [inline skimage/_shared/testing.py]: @hmaarrfk can you add a reference and/or explanation for this?

File: skimage/feature/tests/test_orb.py [hunk 2]
```diff
@@ -69,6 +69,8 @@ def test_keypoints_orb_less_than_desired_no_of_keypoints():
     assert_almost_equal(exp_cols, detector_extractor.keypoints[:, 1])
 
 
+@xfail(condition=arch32,
+       reason='https://github.com/scikit-image/scikit-image/issues/3091')
```
Comment (jni) [inline skimage/feature/tests/test_orb.py]: Here I'd like a more verbose message: "Known test failure on 32-bit platforms. See [link] for details."

**Comment by jni** [comment]
@hmaarrfk thanks!!! This is great. Just two minor comments and :+1: from me!

**Comment by jni** [comment]
In semi-related news, do you know what is happening here?

https://ci.appveyor.com/project/scikit-image/scikit-image-wheels/build/1.0.22/job/twjpd7i85cjdc3pb#L1683

CC @grlee77 

Our windows wheels are failing this test, whether the build is 32 or 64 bit.

**Comment by grlee77** [comment]
> In semi-related news, do you know what is happening here?
>
>https://ci.appveyor.com/project/scikit-image/scikit-image-wheels/build/1.0.22/job/twjpd7i85cjdc3pb#L1683
>

The max level test needs to be updated for PyWavelets 1.0. Exceeding the recommended max level no longer raises a `ValueError`, but instead raises a `UserWarning` (see PyWavelets/pywt#403). I can make a PR to address this later this afternoon/evening.

**Pull request merged**